framework/include/dirackernels/DiracKernel.h

```diff
@@ -51,127 +56,64 @@ class DiracKernel : public ResidualObject,
   /**
    * This gets called by computeOffDiagJacobian() at each quadrature point.
    */
-  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
 
   /**
    * Computes the off-diagonal Jacobian for variable jvar.
    */
   virtual void computeOffDiagJacobian(unsigned int jvar) override;
 
-  virtual const MooseVariableField<Real> & variable() const override { return _var; }
+  virtual const MooseVariableField<T> & variable() const override { return _var; }
 
   /**
    * This is where the DiracKernel should call addPoint() for each point it needs to have a
    * value distributed at.
    */
-  virtual void addPoints() = 0;
-
-  /**
-   * Whether or not this DiracKernel has something to distribute on this element.
-   */
-  bool hasPointsOnElem(const Elem * elem);
-
-  /**
-   * Whether or not this DiracKernel has something to distribute at this Point.
-   */
-  bool isActiveAtPoint(const Elem * elem, const Point & p);
-
-  /**
-   * Remove all of the current points and elements.
-   */
-  void clearPoints();
-
-  /**
-   * Clear point cache when the mesh changes, so that element
-   * coarsening, element deletion, and distributed mesh repartitioning
-   * don't leave this with an invalid cache.
-   */
-  virtual void meshChanged() override;
+  virtual void addPoints() override = 0;
 
 protected:
   /**
    * This is the virtual that derived classes should override for computing the residual.
    */
-  virtual Real computeQpResidual() = 0;
+  virtual Real computeQpResidual() override = 0;
```

Comment:
You can remove these lines as they are in the base class

Reply:
alternatively you could remove the declaration in the base class and keep this one